### PR TITLE
fix outdated contributing to documentation

### LINKED ISSUE #240688: Document CommonMark extensions used in manual
There are syntax elements, like `:::` which are used in the manual but undocumented in the [relevant setion](https://nixos.org/manual/nixpkgs/stable/#chap-contributing).

              the generic syntax is document [here in the nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#sec-contributing-markup), but a few more non-admonition variations have been implemented recently that aren't yet documented in the manual.

_Originally posted by @pennae in https://github.com/nix-community/nixdoc/pull/40#discussion_r1246597078_

### PR DESCRIPTION
**Pull request by mightyiam**
Co-authored-by: Shahar "Dawn" Or <mightyiampresence@gmail.com>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alejandrosame** [comment]
tag:alejandrosamemightyiam

File: doc/contributing/contributing-to-documentation.chapter.md [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
-If you experience problems, run `make debug` to help understand the docbook errors.
+The sources of the Nixpkgs manual are in the [doc](https://github.com/NixOS/nixpkgs/tree/master/doc) subdirectory of the Nixpkgs repository.
 
 After making modifications to the manual, it's important to build it before committing. You can do that as follows:
 
 ```ShellSession
 $ cd /path/to/nixpkgs/doc
 $ nix-shell
-[nix-shell]$ make clean
 [nix-shell]$ nix-build .
```
Comment (jtojnar) [inline doc/contributing/contributing-to-documentation.chapter.md:10]: `nix-shell` call is no longer needed.

Maybe we could just use `nix-build ./doc`.

File: doc/contributing/contributing-to-documentation.chapter.md [hunk 2]
```diff
@@ -29,6 +18,9 @@ As per [RFC 0072](https://github.com/NixOS/rfcs/pull/72), all new documentation
 
 Additional syntax extensions are available, all of which can be used in NixOS option documentation. The following extensions are currently used:
```
Comment (jtojnar) [inline doc/contributing/contributing-to-documentation.chapter.md:20]: Maybe also mention the [other deviations](https://github.com/NixOS/nixpkgs/blob/8b1253715d80e224908e29a53014b9323933abf7/pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/md.py#L581-L608):

- HTML is not allowed
- [Examples](https://github.com/NixOS/nixpkgs/blob/be4d19ff1a9a327ae805fdb344470ed6450256fc/doc/stdenv/stdenv.chapter.md?plain=1#L467-L482)
- [Figures](https://github.com/NixOS/nixpkgs/blob/be4d19ff1a9a327ae805fdb344470ed6450256fc/doc/contributing/submitting-changes.chapter.md?plain=1#L217-L221)
- [Footnotes](https://github.com/NixOS/nixpkgs/commit/538b3d1b3c12cc07f00dcec374b16a469b3679ff)

File: doc/contributing/contributing-to-documentation.chapter.md [hunk 3]
```diff
@@ -29,6 +18,9 @@ As per [RFC 0072](https://github.com/NixOS/rfcs/pull/72), all new documentation
...
 
+- []{#ssec-contributing-markup-tables}
+  [GFM tables](https://github.github.com/gfm/#tables-extension-).
```
Comment (jtojnar) [inline doc/contributing/contributing-to-documentation.chapter.md:22]:
```suggestion
  Tables, using the [GitHub-flavored Markdown syntax](https://github.github.com/gfm/#tables-extension-).
```

**Review by pennae — COMMENTED** [review]
jtojnar got most of it, but there's a few more things:

- `{=include=}` code blocks in the `manual` processor
  - `{=include=} options` is a bit of a beast since it interacts with `nixosOptionsDoc`
- headings in examples and figures create anchors, not just a title caption
- the `[Ctrl]{.keycap}` span was apparently never documented

also not quite sure we can still reasonably call our docs language CommonMark at this point, it's much closer to MyST now?

**Comment by alejandrosame** [comment]
@jtojnar and @pennae: Thank you for your comments! I created a tracking issue (https://github.com/NixOS/nixpkgs/issues/246357) to have a better overview of what would be required to properly reference the syntax supported by `nixos-render-docs`.

I had in mind both keeping your feedback and preventing this PR from staying open for too long. Taking smaller steps already improves the situation we are currently in.

Please add further comments as you see fit in the issue. Without going too deep yet into what's supported, it was not obvious how the list of currently documented features maps into `nixos-render-docs` source code. I'll deal with it as I document syntax features.

**Review by infinisil — APPROVED** [review]
Looking good to me! Let's not make this include too many things.

**Pull request merged**